sylvia-derive/src/contract/communication/reply.rs

```diff
@@ -0,0 +1,660 @@
+
+        if self.payload.len() != new_reply_data.payload.len() {
+            emit_error!(current_method_name.span(), "Mismatched quantity of method parameters.";
+                note = self.handler_id.span() => format!("Both `{}` handlers should have the same number of parameters.", self.handler_id);
+                note = new_handler.function_name().span() => format!("Previous definition of {} handler.", self.handler_id)
+            );
+        }
+
+        self.payload
+            .iter()
+            .zip(new_reply_data.payload.iter())
+            .for_each(|(current_field, new_field)|
+        {
+            if current_field.ty() != new_field.ty() {
+                emit_error!(current_field.name().span(), "Mismatched parameter in reply handlers.";
+                    note = current_field.name().span() => format!("Parameters for the `{}` handler have to be the same.", self.handler_id);
+                    note = new_field.name().span() => format!("Previous parameter defined for the `{}` handler.", self.handler_id)
+                )
+            }
+        });
+
+        let new_function_name = new_handler.function_name();
+        let new_reply_on = new_handler.msg_attr().reply_on();
+        self.handlers.push((new_function_name, new_reply_on));
+    }
+
+    /// Emits success and error match arms for a single `ReplyId`.
+    fn emit_match_arms(&self, contract: &Type, generics: &[&GenericParam]) -> TokenStream {
+        let reply_id = &self.reply_id;
+        let contract_turbofish = emit_turbofish(contract, generics);
+        let success_match_arm = self.emit_success_match_arm(&contract_turbofish);
+        let error_match_arm = self.emit_error_match_arm(&contract_turbofish);
+
+        quote! {
+            #reply_id => {
+                match result {
+                    #success_match_arm
+                    #error_match_arm
+                }
+            }
+        }
+    }
+
+    /// Emits [cosmwasm_std::ReplyOn] value to be put in the `cosmwasm_std::SubMsg`.
+    /// If both `Success` and `Error` is defined for given `reply_id`, `cosmwasm_std::ReplyOn::Always` is returned.
+    fn emit_cw_reply_on(&self) -> TokenStream {
+        let sylvia = crate_module();
+        let is_always = self
+            .handlers
+            .iter()
+            .any(|(_, reply_on)| reply_on == &ReplyOn::Always);
+        let is_success = self
+            .handlers
+            .iter()
+            .any(|(_, reply_on)| reply_on == &ReplyOn::Success);
+        let is_error = self
+            .handlers
+            .iter()
+            .any(|(_, reply_on)| reply_on == &ReplyOn::Error);
+
+        if is_always || (is_success && is_error) {
+            quote! { #sylvia ::cw_std::ReplyOn::Always }
+        } else if is_success {
+            quote! { #sylvia ::cw_std::ReplyOn::Success }
+        } else if is_error {
+            quote! { #sylvia ::cw_std::ReplyOn::Error }
+        } else {
+            // This should never happen.
```

Comment:
If this case will never happen, why don't we go for something like:
```rust
if is_always || (is_success && is_error) {
    quote! { #sylvia ::cw_std::ReplyOn::Always }
} else if is_success {
    quote! { #sylvia ::cw_std::ReplyOn::Success }
} else {
    quote! { #sylvia ::cw_std::ReplyOn::Error }
}
```

Reply:
you are right, we can